### ISSUE TITLE
host-local plugin only implements CNI 0.3.1 instead of 0.4.0

### ISSUE THREAD

**Issue by weibeld** [containernetworking/plugins#425]
Installed Kubernetes v1.17.0 with kubeadm v1.17.0. The installed version of the `host-local` plugin only implements version 0.3.1 of the CNI specification:

```
$ CNI_COMMAND=VERSION /opt/cni/bin/host-local <host-local.conf
{"cniVersion":"0.3.1","supportedVersions":["0.1.0","0.2.0","0.3.0","0.3.1"]}
```

So, if you call `host-local` from another CNI plugin which implements version 0.4.0 and also assume version 0.4.0 for the `host-local` plugin, it results in a version mismatch error:

```
$ /opt/cni/bin/host-local <host-local.conf
{
    "code": 1,
    "msg": "incompatible CNI versions",
    "details": "config is \"0.4.0\", plugin supports [\"0.1.0\" \"0.2.0\" \"0.3.0\" \"0.3.1\"]"
}
```

Where `host-local.conf` might look like this:

```json
{
  "cniVersion": "0.4.0",
  "name": "my-pod-network",
  "ipam": {
    "type": "host-local",
    "ranges": [ [ { "subnet": "20.0.0.0/24" } ] ]
  }
}
```

The `host-local` plugin should be updated to support version 0.4.0 of the CNI specification.

Other plugins might also be stuck at 0.3.1, see, for example, #359.

**Comment by squeed** [comment]
@weibeld The problem is just that kubeadm bundles a very old CNI package. Host-local supports v0.4.0 and even implements check :-). 

```console
$ ./build_linux.sh && CNI_COMMAND=VERSION ./bin/host-local
{"cniVersion":"0.4.0","supportedVersions":["0.1.0","0.2.0","0.3.0","0.3.1","0.4.0"]}
```

I'm not familiar with the kubeadm governance; would you mind filing an issue for them? Thanks.

**Comment by weibeld** [comment]
@squeed Thanks a lot for this information! I just created an [issue](https://github.com/kubernetes/kubeadm/issues/1976) in the kubeadm repo.

**Comment by weibeld** [comment]
For the records:

kubeadm uses the default kubelet configuration defined by the current version of Kubernetes, and the current version (Kubernetes v1.17) uses [CNI plugins v0.7.5](https://github.com/kubernetes/release/blob/e29fdb0c249a44e9e1cc3f0507de65a770c8bf44/build/rpms/kubelet.spec#L14). These plugins only support CNI spec 0.3.1.

There is a [pull request](https://github.com/kubernetes/release/pull/731/files) to update the CNI plugins package to the newest version v0.8.3, however it is on hold and [scheduled for Kubernetes v1.18](https://github.com/kubernetes/release/pull/731#issuecomment-558992751).

So, I guess that the newest CNI plugins supporting CNI 0.4.1 will be included automatically in Kubernetes 1.18.
